DQM/HLTEvF/plugins/LumiMonitor.cc

```diff
@@ -0,0 +1,212 @@
+  lumiVsLS_ = ibooker.bookProfile(histname, histtitle, 
+				  lsPSet.getParameter<int32_t>("nbins"), 0., double(lsPSet.getParameter<int32_t>("nbins")),
+				  lumiPSet.getParameter<double>("xmin"), lumiPSet.getParameter<double>("xmax"));
+  //  lumiVsLS_->getTH1()->SetCanExtend(TH1::kAllAxes);
+  lumiVsLS_->setAxisTitle("LS",1);
+  lumiVsLS_->setAxisTitle("(SCAL) inst lumi E30 [Hz cm^{-2}]",2);
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void LumiMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  //  int bx = iEvent.bunchCrossing();
+  int ls = iEvent.id().luminosityBlock();
+
+  float scal_lumi = -1.;
+  edm::Handle<LumiScalersCollection> lumiScalers;
+  iEvent.getByToken(lumiscalersToken_, lumiScalers);
+  if ( lumiScalers.isValid() && lumiScalers->size() ) {
+    LumiScalersCollection::const_iterator scalit = lumiScalers->begin();
+    scal_lumi = scalit->instantLumi();
+  } else {
+    scal_lumi = -1.;
+  }
+  lumiVsLS_ -> Fill(ls, scal_lumi);
+
+  if ( doPixelLumi_ ) {
+    size_t pixel_clusters = 0;
+    float  pixel_lumi = -1.;
+    edm::Handle< edmNew::DetSetVector<SiPixelCluster> > pixelClusters;
+    iEvent.getByToken(pixelClustersToken_, pixelClusters);
+    if ( pixelClusters.isValid() ) {
+      
+      edm::ESHandle<TrackerTopology> tTopoHandle;
+      iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
+      const TrackerTopology* const tTopo = tTopoHandle.product();
+      
+      // Count the number of clusters with at least a minimum
+      // number of pixels per cluster and at least a minimum charge.
+      size_t tot = 0;
+      edmNew::DetSetVector<SiPixelCluster>::const_iterator  pixCluDet = pixelClusters->begin();
+      for ( ; pixCluDet!=pixelClusters->end(); ++pixCluDet) {
+	
+	DetId detid = pixCluDet->detId();
+	size_t subdetid = detid.subdetId();
+	if ( subdetid == (int) PixelSubdetector::PixelBarrel ) 
+	  if ( tTopo->layer(detid)==1 ) 
+	    continue;
+	
+	edmNew::DetSet<SiPixelCluster>::const_iterator  pixClu = pixCluDet->begin();    
+	for ( ; pixClu != pixCluDet->end(); ++pixClu ) {
+	  ++tot;
+	  if ( (pixClu->size()   >= minNumberOfPixelsPerCluster_) &&
+	       (pixClu->charge() >= minPixelClusterCharge_      ) ) {
+	    ++pixel_clusters;
+	  }
+	}
+      }
+      pixel_lumi = lumi_factor_per_bx_ * pixel_clusters / GetLumi::CM2_TO_NANOBARN ; // ?!?!
+    } else
+      pixel_lumi = -1.;
+
+    numberOfPixelClustersVsLS_   -> Fill(ls,       pixel_clusters);
+    numberOfPixelClustersVsLumi_ -> Fill(scal_lumi,pixel_clusters);
+    pixelLumiVsLS_               -> Fill(ls,       pixel_lumi);
+    pixelLumiVsLumi_             -> Fill(scal_lumi,pixel_lumi);
+  }
+
+}
+
+void LumiMonitor::fillHistoPSetDescription(edm::ParameterSetDescription & pset)
```

Comment:
Is there a particular reason to modify an existing `edm::ParameterSetDescription`, rather than creating a new one ?
This way you could slightly simplify the code that uses `fillHistoPSetDescription`.